projects/vdk-core/src/vdk/api/plugin/connection_hook_spec.py

```diff
@@ -68,6 +70,54 @@ def db_connection_decorate_operation(
         """
         pass
 
+    @hookspec(firstresult=True)
+    def db_connection_execute_operation(
+        self, raw_cursor: Any, managed_operation: ManagedOperation
```

Comment:
`raw_cursor: Any` contrasts with `managed_operation: ManagedOperation` abstraction,
I'll elaborate  further

Reply:
raw_cursor exposed is an anti-pattern in a use-case where we strive being on the data path

Reply:
I've created new ExecutionCursor and documented there how it should be used. It has much more clear contract this way.